application/core/Traits/LSApplicationTrait.php

```diff
@@ -8,17 +8,16 @@
 
 trait LSApplicationTrait
 {
-
     /* @var integer| null the current userId for all action */
     private $currentUserId;
     /**
      * get the current id of connected user,
      * check if user exist before return for security
-     * @return int|null user id
+     * @return int|null user id, 0 mean invalid user
      */
     public function getCurrentUserId()
     {
-        if(empty(App()->session['loginID'])) {
+        if (empty(App()->session['loginID'])) {
```

Comment:
@gabrieljenik empty or is_null here ?

Reply:
I would say empty better. But why the doubt? Which edge case are you thinking about?

Reply:
I really don't know who wan use `App()->session['loginID'] = 0` or `App()->session['loginID'] = ''` ?

We currently use loginID in all test and set it when login. But it's old version, when we don't use Yii …

Reply:
I like empty more.

Reply:
> I like empty more. Catches more situtations.

Yes, 
In all current code code : empty App()->session['loginID'] is always null. Then no issue. I just think of plugin or hacked version using 0 or ''
Here : potential issue about loop.